app/helpers/application_helper/button/miq_action_modify.rb

```diff
@@ -1,23 +1,32 @@
 class ApplicationHelper::Button::MiqActionModify < ApplicationHelper::Button::Basic
+  def role_allows_feature?
+    super && role_allows?(:feature => 'event_edit')
+  end
+
+  def visible?
+    @view_context.x_active_tree != :event_tree
+  end
+
   def disabled?
-    @view_context.x_node.split("_").any? do |level|
-      node_type, id = level.split('-')
-      node_type == 'p' && MiqPolicy.find_by(:id => id).try(:read_only)
+    if any_policy_read_only?
+      @error_message = case policy_type
+                       when 'a'  then _('This Action belongs to a read only Policy and cannot be modified')
+                       when 'ev' then _('This Event belongs to a read only Policy and cannot be modified')
+                       end
     end
+    @error_message.present?
   end
 
-  def calculate_properties
-    super
+  private
 
-    if disabled?
-      self[:title] = case @view_context.x_node.split("_").last.split('-')[0]
-                     when 'a'  then _("This Action belongs to a read only Policy and cannot be modified")
-                     when 'ev' then _("This Event belongs to a read only Policy and cannot be modified")
-                     end
+  def any_policy_read_only?
+    @view_context.x_node.split('_').any? do |level|
+      node_type, id = level.split('-')
+      node_type == 'p' && MiqPolicy.find(id).try(:read_only)
     end
   end
 
-  def visible?
-    @view_context.x_active_tree != :event_tree && role_allows?(:feature => 'event_edit')
+  def policy_type
+    @view_context.x_node.split('_').last.split('-')[0]
   end
 end
```

Comment:
I'm getting error for this button:
```
[----] F, [2016-12-09T13:40:36.220929 #13128:2ac71c77a7a0] FATAL -- : Error caught: [ActiveRecord::RecordNotFound] Couldn't find MiqPolicy with 'id'=10r8
/home/rblanco/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/activerecord-5.0.0.1/lib/active_record/core.rb:173:in `find'
/home/rblanco/devel/manageiq/app/helpers/application_helper/button/miq_action_modify.rb:25:in `block in any_policy_read_only?'
/home/rblanco/devel/manageiq/app/helpers/application_helper/button/miq_action_modify.rb:23:in `any?'
/home/rblanco/devel/manageiq/app/helpers/application_helper/button/miq_action_modify.rb:23:in `any_policy_read_only?'
/home/rblanco/devel/manageiq/app/helpers/application_helper/button/miq_action_modify.rb:11:in `disabled?'
/home/rblanco/devel/manageiq/app/helpers/application_helper/button/basic.rb:46:in `calculate_properties'
/home/rblanco/devel/manageiq/app/helpers/application_helper/toolbar_builder.rb:176:in `apply_common_props'
/home/rblanco/devel/manageiq/app/helpers/application_helper/toolbar_builder.rb:88:in `toolbar_button'
/home/rblanco/devel/manageiq/app/helpers/application_helper/toolbar_builder.rb:112:in `block in build_select_button'
/home/rblanco/devel/manageiq/app/helpers/application_helper/toolbar_builder.rb:106:in `each'
/home/rblanco/devel/manageiq/app/helpers/application_helper/toolbar_builder.rb:106:in `each_with_index'
/home/rblanco/devel/manageiq/app/helpers/application_helper/toolbar_builder.rb:106:in `build_select_button'
/home/rblanco/devel/manageiq/app/helpers/application_helper/toolbar_builder.rb:243:in `build_button'
/home/rblanco/devel/manageiq/app/helpers/application_helper/toolbar_builder.rb:29:in `block (2 levels) in build_toolbar'
/home/rblanco/devel/manageiq/app/helpers/application_helper/toolbar_builder.rb:28:in `each'
/home/rblanco/devel/manageiq/app/helpers/application_helper/toolbar_builder.rb:28:in `block in build_toolbar'
/home/rblanco/devel/manageiq/app/helpers/application_helper/toolbar_builder.rb:21:in `each'
/home/rblanco/devel/manageiq/app/helpers/application_helper/toolbar_builder.rb:21:in `each_with_index'
/home/rblanco/devel/manageiq/app/helpers/application_helper/toolbar_builder.rb:21:in `build_toolbar'
/home/rblanco/devel/manageiq/app/helpers/application_helper/toolbar_builder.rb:6:in `call'
/home/rblanco/devel/manageiq/app/helpers/application_helper.rb:378:in `build_toolbar'
/home/rblanco/devel/manageiq/app/controllers/miq_policy_controller.rb:495:in `replace_right_cell'
/home/rblanco/devel/manageiq/app/controllers/miq_policy_controller.rb:269:in `tree_select'
```

![screenshot from 2016-12-09 13-42-26](https://cloud.githubusercontent.com/assets/1187051/21049485/6180c93e-be15-11e6-85db-33581cb7c4b1.png)